### PR TITLE
feat(d11):control sending screenshots with each crash type

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change

1. control sending screenshots with each crash type

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

JIRA ID :

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by AndrewAminInstabug — APPROVED** [review]

File: src/models/ReproConfig.ts [hunk 1]
```diff
@@ -12,8 +12,10 @@ export interface ReproConfig {
    * Repro steps mode for Crash Reporting.
    *
    * @default ReproStepsMode.enabledWithNoScreenshots
+   *
+   * When this is set, `anr`, `appHangs`,`fatalCrash`,`nonFatalCrash`,`forceRestart`, and `outOfMemory` will be ignored.
    */
-  crash?: ReproStepsMode;
+  allCrashes?: ReproStepsMode;
```
Comment (a7medev) [inline src/models/ReproConfig.ts:18]: We can't remove the old `crash` name as that would be a breaking change, we can either keep it `crash` as is or keep it with a `@deprecated` flag and add the new `allCrashes` one then remove `@deprecated` in a future major release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## [13.0.6](https://github.com/Instabug/Instabug-React-Native/compare/v13.0.4...v13.0.5) (JUN 15, 2024)
```
Comment (a7medev) [inline CHANGELOG.md:3]:
```suggestion
## [Unreleased](https://github.com/Instabug/Instabug-React-Native/compare/v13.0.5...dev)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+
+### Changed
+
+- support sending screenshot for `ANR`, `appHangs`,`fatal-Crash`,`non-Fatal-Crash`,`force-Restart`,
+  and `out-of-Memory` crash types using `Instabug.setReproStepsConfig` API.
```
Comment (a7medev) [inline CHANGELOG.md:8]:
```suggestion
- Support controlling repro steps for ANRs, app hangs, fatal crashes, non-fatal crashes, force restarts, and out-of-memory crash types separately using the `Instabug.setReproStepsConfig` API.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+- Bump Instabug Android SDK to v13.0.6 . [See release notes](https://github.com/Instabug/android/releases/tag/v13.0.6).
+- Bump Instabug iOS SDK to v13.0.6 . [See release notes](https://github.com/instabug/instabug-ios/releases/tag/13.0.6).
+
```
Comment (a7medev) [inline CHANGELOG.md:11]: Should these changes be part of this PR?

File: examples/default/src/App.tsx [hunk 5]
```diff
@@ -22,10 +17,16 @@ export const App: React.FC = () => {
...
-      all: ReproStepsMode.enabled,
+      bug: ReproStepsMode.enabledWithNoScreenshots,
+      allCrashes: ReproStepsMode.enabledWithNoScreenshots,
+      anr: ReproStepsMode.enabledWithNoScreenshots,
+      appHangs: ReproStepsMode.enabledWithNoScreenshots,
+      sessionReplay: ReproStepsMode.enabledWithNoScreenshots,
+      fatalCrash: ReproStepsMode.enabledWithNoScreenshots,
+      nonFatalCrash: ReproStepsMode.enabledWithNoScreenshots,
+      forceRestart: ReproStepsMode.enabledWithNoScreenshots,
+      outOfMemory: ReproStepsMode.enabled,
```
Comment (a7medev) [inline examples/default/src/App.tsx:29]: We have some options overriding others here which can cause some confusion, let's probably keep all enabled as it was or use a config with no conflicting options if needed.

File: src/models/ReproConfig.ts [hunk 6]
```diff
@@ -22,10 +24,51 @@ export interface ReproConfig {
...
+   *
+   * @default ReproStepsMode.enabled
+   */
+  forceRestart?: ReproStepsMode;
+
+  /**
+   * Repro steps mode for outOfMemory crashes on ios.
+   *
+   * @default ReproStepsMode.enabled
+   */
```
Comment (a7medev) [inline src/models/ReproConfig.ts:65]: Are we sure all these options default to `ReproStepsMode.enabled`? I'd think they should default to `ReproStepsMode.enabledWithNoScreenshots` as crashes, can you please confirm?

**Review by a7medev — CHANGES_REQUESTED** [review]
Happy Eid guys! ❤️

I have a few comments here, can you please check them before merging this PR? 🙏🏼

Also, there are some changes which I think are irrelevant to this PR like the changes related to versioning and releases, do we need these here?

**Review by mzelzoghbi — APPROVED** [review]

**Pull request merged**